web/src/engine/websites/CuuTruyen.ts

```diff
@@ -0,0 +1,113 @@
+﻿import { Tags } from '../Tags';
+import icon from './CuuTruyen.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as W } from '../../i18n/ILocale';
+
+type APIResult<T> = {
+    data: T,
+    _metadata: {
+        total_pages: number,
+    }
+}
+
+type APIManga = {
+    id: number,
+    name: string
+}
+
+type APIChapter = {
+    id: number,
+    name: string,
+    number: string,
+    status: string,
+    pages: {
+        image_url: string,
+        drm_data: string,
+        status: string
+    }[]
+}
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = `${this.URI.origin}/api/v2/`;
+
+    public constructor() {
+        super('cuutruyen', 'Cứu Truyện', 'https://cuutruyen.net', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Vietnamese, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/mangas/\\d+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = url.match(/\/mangas\/([0-9]+)/)[1];
+        const { data: { name } } = await FetchJSON<APIResult<APIManga>>(new Request(new URL(`mangas/${mangaid}`, this.apiUrl)));
+        return new Manga(this, provider, mangaid, name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
```

Comment:
In case total pages are known in advance and the website has a less aggressive rate-limit, a task-pool could be used to parallelize the requests for possible speed improvement e.g., see _ReaperScans_ or _Piccoma_

Reply:
Changed it but i see no improvements, perhaps its my rate limit fault?

Reply:
I checked the website and there are only 36 index pages, which is not worth using a taskpool.
Took me less than 4 seconds to download the manga list.